src/daemon/absorber.cc

```diff
@@ -67,6 +67,15 @@ bool Absorber::HandleNewMessage(net::ConnectionPtr connection,
 
   if (message->HasExtension(proto::Remote::extension)) {
     Message execute(message->ReleaseExtension(proto::Remote::extension));
+    auto config = conf();
+    if (execute->emitter_version() < config->absorber().min_emitter_version()) {
+      LOG(WARNING) << "Emitter sent a bad version: "
```

Comment:
Better not to mention `emitter` and just say about "incompatible" or "deprecated" client version - not the "bad" one. Also it's better to sync wording with the new name of the Status enum.